src/gax/src/response.rs

```diff
@@ -235,13 +247,15 @@ impl<T> Response<T> {
 ///     Some(&http::HeaderValue::from_static("application/json"))
 /// );
 /// ```
-///  
+///
 /// [tower]: https://github.com/tower-rs/tower
 #[derive(Clone, Debug, Default)]
 #[non_exhaustive]
 pub struct Parts {
     /// The HTTP headers or the gRPC metadata converted to HTTP headers.
     pub headers: http::HeaderMap<http::HeaderValue>,
+    /// Extensions for passing data from the transport.
+    pub extensions: Arc<Mutex<http::Extensions>>,
```

Comment:
If we decide to go with this approach we should link a bug / document describing the tradeoffs and why we decided on this.